examples/regression/script.jl

```diff
@@ -0,0 +1,196 @@
+# where `u ~ f(z)` are the pseudo-points. We parameterise the covariance matrix
+# of `q` as `C = AᵀA` since this guarantees that `C` is positive definite.
+
+m_init = zeros(M)
+A_init = Matrix{Float64}(I, M, M)
+q_init = MvNormal(m_init, A_init'A_init)
+#md nothing #hide
+
+# Given a set of parameters, we now define a Flux 'layer' which forms the basis
+# of our model.
+
+struct SVGPModel
+    k  # kernel parameters
+    z  # inducing points
+    m  # variational mean
+    A  # variational covariance
+end
+
+Flux.@functor SVGPModel (k, z, m, A)
+#md nothing #hide
+
+# Create the 'model' from the parameters - i.e. return the FiniteGP at inputs x,
+# the FiniteGP at inducing inputs z and the variational posterior over inducing
+# points - q(u).
+
+lik_noise = 0.3
+jitter = 1e-5
+
+# Next, we define some useful functions on the model - creating the prior GP
+# under the model, as well as the `SVGP` struct needed to create the posterior
+# approximation and to compute the ELBO.
+
+function prior(m::SVGPModel)
+    kernel = make_kernel(m.k)
+    return GP(kernel)
+end
+
+function make_approx(m::SVGPModel, prior)
+    q = MvNormal(m.m, m.A'm.A)
+    fz = prior(m.z, jitter)
+    return SVGP(fz, q)
+end
+#md nothing #hide
+
+# Create the approximate posterior GP under the model.
+
+function model_posterior(m::SVGPModel)
+    svgp = make_approx(m, prior(m))
+    return posterior(svgp)
+end
+#md nothing #hide
+
+# Define a predictive function for the model - in this case the prediction is
+# the joint distribution of the approximate posterior GP at some test inputs `x`
+# (defined by an `AbstractGPs.FiniteGP`).
+
+function (m::SVGPModel)(x)
+    post = model_posterior(m)
+    return post(x)
+end
+#md nothing #hide
+
+# Return the loss given data - for the SVGP, the loss used is the negative ELBO
+# (also known as the Variational Free Energy). `n_data` is required for
+# minibatching used below.
+
+function loss(m::SVGPModel, x, y; n_data=length(y))
+    f = prior(m)
+    fx = f(x, lik_noise)
+    svgp = make_approx(m, f)
+    return -elbo(svgp, fx, y; n_data)
+end
+#md nothing #hide
+
+# Finally, create the model with initial parameters
+
+model = SVGPModel(k_init, z_init, m_init, A_init)
+#md nothing #hide
+
+# ## Training the model
+#
+# Training the model now simply proceeds with the usual `Flux.jl` training loop.
+
+opt = ADAM(0.001)  # Define the optimiser
+params = Flux.params(model)  # Extract the model parameters
+#md nothing #hide
+
+# One of the major advantages of the SVGP model is that it allows stochastic
+# estimation of the ELBO by using minibatching of the training data. This is
+# very straightforward to achieve with `Flux.jl`'s utilities:
+
+b = 100 # minibatch size
+data_loader = Flux.Data.DataLoader((x, y); batchsize=b)
+
+# The loss (negative ELBO) before training
+
+println(loss(model, x, y))
+
+# Train the model
+
+Flux.train!(
+    (x, y) -> loss(model, x, y; n_data=N),
```

Comment:
Could you make more explicit that, when minibatching, it's crucial to explicitly pass the full length of the dataset (so that the scaling of data-fit vs KL penalty terms is correct)? maybe a comment here and a mention in the text above?